flutter/lib/src/navigation/sentry_navigation_observer.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/widgets.dart';
+import 'package:sentry/sentry.dart';
+
+import 'navigation_breadcrumb.dart';
+
+/// This is a navigation observer to record navigational breadcrumbs.
+/// For now it only records navigation events and no gestures.
+///
+/// Routes can always be null and their settings can also always be null.
+/// For example, if the application starts, there is no previous route.
+/// The RouteSettings are null if a developer has not specified any
+/// RouteSettings.
+///
+/// See also:
+///   - https://api.flutter.dev/flutter/widgets/RouteObserver-class.html
+class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
+  factory SentryNavigatorObserver(Hub hub) {
+    if (hub != null) {
+      return SentryNavigatorObserver._(hub);
+    }
+    return SentryNavigatorObserver._(Sentry.currentHub);
+  }
```

Comment:
the idea here would be using an `Integration` (default_integrations.dart) that takes a `Hub` and `SentryOptions`,
also adding a new `bool` to `SentryOptions` so we can turn it off if necessary.
Then adding the `Integration` at `_addDefaultIntegrations` (sentry_flutter.dart)

See how we do on Android:

https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/ActivityBreadcrumbsIntegration.java

Reply:
A navigation observer must unfortunately be added manually as they can't be added automatically.
See https://api.flutter.dev/flutter/material/MaterialApp/navigatorObservers.html

´´´dart
MaterialApp(
   navigatorObservers: [
     SentryNavigatorObserver(),
   ],
   // other parameter ...
)
´´´

So I guess the `Integration` and the `SentryOptions` option are unnecessary?
I would however add this to the readme so users are aware of it.

Reply:
I guess the `ActivityBreadcrumbsIntegration.java` is more equivalent to the [`WidgetsBindingObserver`](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html) (this would be a nice addition in another PR) than the `NavigationObserver`.

Reply:
> I guess the `ActivityBreadcrumbsIntegration.java` is more equivalent to the [`WidgetsBindingObserver`](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html) (this would be a nice addition in another PR) than the `NavigationObserver`.

true, I was just trying to give examples on how to use the `Integration` feature

Reply:
Ah, I see. So what do you suggest on how to make the `SentryNavigationObserver` available to users?

Reply:
> A navigation observer must unfortunately be added manually as they can't be added automatically.
> See https://api.flutter.dev/flutter/material/MaterialApp/navigatorObservers.html
> 
> ```dart
> MaterialApp(
>    navigatorObservers: [
>      SentryNavigatorObserver(),
>    ],
>    // other parameter ...
> )
> ```
> 
> So I guess the `Integration` and the `SentryOptions` option are unnecessary?
> I would however add this to the readme so users are aware of it.

Oh I see, that makes sense, yeah if the user needs to add manually, then for sure `Integration` and `SentryOptions` are unnecessary, thanks for pointing this out.

Reply:
> Ah, I see. So what do you suggest on how to make the `SentryNavigationObserver` available to users?

reading the `NavigatorObserver` docs now I see that it'd not be possible to hook it up automatically unless we create our own `SentryMaterialApp` but I guess this is too much intrusive and if people need to create a new `SentryMaterialApp` anyway, I'd rather having less code and documenting they should do (what u have done on the README), so you are on the right path.

Reply:
thanks for clarifying btw, I didn't play enough with the Flutter UI yet.